src/app/(profile)/(components)/TogglePublishButton/TogglePublishButton.tsx

```diff
@@ -23,10 +24,17 @@ export const TogglePublishButton = (props: TogglePublishButtonProps) => {
   const { loading, runAsync } = useAsyncAction()
 
   const handleButtonClick = async () => {
-    await runAsync(async () => {
-      await publishProfile(profileId)
-      setShowPopup(true)
-    })
+    if (state == PublishingState.APPROVED) {
```

Comment:
**issue:** This will not work correctly as we have also `PublishingState.REJECTED`. Let's change this function to:

```ts
const handleButtonClick = async () => {

  switch (state) {
      case PublishingState.APPROVED:
       await runAsync(async () => {
        await unpublishProfile(profileId)
        setShowPopup(true)
      })
      return
      case PublishingState.DRAFT:
       await runAsync(async () => {
        await publishProfile(profileId)
        setShowPopup(true)
      })
      return
      case PublishingState.REJECTED:
       await runAsync(async () => {
        await publishProfile(profileId)
        setShowPopup(true)
      })
      return
      default:
      throw Error("Wrong profile action")
   }

}
```